### PR TITLE
Added target validation for job API

### PR DESCRIPTION
**Pull request by yhwen**
Fixes # .

### Description

- Added target validation for job API, also added the unit test.

1. must provide a target name
2. target name can not use special character

 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: tests/unit_test/job_config/fed_job_test.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+class TestFedJob:
+    def test_validate_targets(self):
+        job = FedJob()
+        component = FedAvg()
+
+        job.to(component, "server")
+        job.to(component, f"site-1")
+
+        with pytest.raises(Exception):
```
Comment (holgerroth) [inline tests/unit_test/job_config/fed_job_test.py:29]: Should we test the target=None as well?

Comment (yhwen) [reply tests/unit_test/job_config/fed_job_test.py:29]: Added the unit test for this case.

**Review by holgerroth — APPROVED** [review]
Looks good.

File: nvflare/job_config/fed_job.py [hunk 2]
```diff
@@ -265,6 +269,14 @@ def simulator_run(self, workspace, threads: int = None):
             gpu=",".join([self._gpus[client] for client in self._gpus.keys()]),
         )
 
+    def _validate_target(self, target):
```
[the remark this answers is not in the document]
Comment (yhwen) [reply nvflare/job_config/fed_job.py:272]: We do allow the admin to use "@", however, we don't allow "@" as a site-name. The "@" character may cause several potential site job execution issue.

**Pull request merged**